gravitee-cockpit-connectors-ws/src/main/java/io/gravitee/cockpit/connectors/ws/channel/ClientChannel.java

```diff
@@ -135,15 +143,29 @@ void listen() {
                         Command<?> command = Json.decodeValue(incoming.replace(COMMAND_PREFIX, ""), Command.class);
                         CommandHandler<Command<?>, Reply> commandHandler = commandHandlers.get(command.getType());
 
+                        Single<Command<?>> commandHandlingObs = Single
+                            .just(command)
+                            .flatMap(
+                                cmd -> {
+                                    if (cmd.getType().equals(Type.GOODBYE_COMMAND)) {
+                                        return handleGoodbye(cmd);
+                                    }
+                                    return Single.just(cmd);
+                                }
+                            );
+
```

Comment:
Why not just something like:
```suggestion
if (cmd.getType().equals(Type.GOODBYE_COMMAND)) {
     handleGoodbye(cmd);
}
```

Reply:
Because at first I didn't known what I was doing :)